### ISSUE TITLE
Incorrect module field in package.json and missing output in ESM format

### ISSUE THREAD

**Issue by Lxxyx** [blitz-js/superjson#126]
Version: superjson@1.6.2

## Problem
The module field is specified in package.json, but the file does not exist. https://github.com/blitz-js/superjson/blob/f910c7b6fc20aa2e0eccdc39a269ba8657ce54a6/package.json#L49-L49

This affects ESM-enabled packaging tools like Vite.

```
> error: [vite:dep-pre-bundle] Failed to resolve entry for package "superjson". The package may have incorrect main/module/exports specified in its package.json.
```

## Expected
Output ESM format with correct module fields

**Comment by Skn0tt** [comment]
Ahh yes! Do you want to provide a PR to fix this?

**Comment by Lxxyx** [comment]
> Ahh yes! Would you be open to provide a PR to fix this?

Sure, but I'd like to know why `tsc` is used instead of `tsdx build`. `tsdx build` can already output code in ESM format.

**Comment by flybayer** [comment]
I recommend tsdx

**Comment by Skn0tt** [comment]
> I recommend tsdx

Could you elaborate on why? We used to use tsdx for SuperJSON, but I couldn't see the benefits over `tsc`, really.

**Comment by Skn0tt** [comment]
Published in v1.7.3
